### ISSUE TITLE
connect with 3rd party OpenAI compatible API

### ISSUE THREAD

**Issue by eschmidbauer** [kudoai/chatgpt.js#80]
vLLM supports an OpenAI-Compatible API. is it possible to configure this code to connect to another service other than OpenAI? 
I've tried modifying the code but i am getting errors.
![image](https://github.com/kudoai/chatgpt.js/assets/7139998/7757e9cb-6c6d-4c80-83b5-8036ffe72697)

**Comment by adamlui** [comment]
Yes it should be possible @eschmidbauer , can you show your code?

**Comment by eschmidbauer** [comment]
basically just these two changes and it doesn't seem to work. 

![image](https://github.com/kudoai/chatgpt.js/assets/7139998/3b640e3e-15ae-4937-9db3-59bd788cbcfc)
![Screenshot 2023-07-17 at 12 28 54 PM](https://github.com/kudoai/chatgpt.js/assets/7139998/08d97144-4627-4173-a054-569115c32001)

**Comment by adamlui** [comment]
Also in the manifest.content_scripts.matches you should add the vLLM site URL. Could you share it with me so I can test making it work by modifying chatgpt.js if necessary?

**Comment by eschmidbauer** [comment]
sorry it's a private URL- i will update the `manifest.content_scripts.matches` file and test again

**Comment by eschmidbauer** [comment]
![Screenshot 2023-07-19 at 12 38 27 PM](https://github.com/kudoai/chatgpt.js/assets/7139998/ecd66c46-06c0-4433-8de8-73c9d93149be)
same issue

**Comment by adamlui** [comment]
What's the errors in console say? also vLLM probably accepts different format payloads than OpenAI ones, so I would need to study their API to add a method to use theirs

**Comment by eschmidbauer** [comment]
it's the same API spec as OpenAI's
I ended up running streamlit to implement a chat interface using the openai package

**Comment by adamlui** [comment]
For ChatGPT.js it started off as a library of functions to interact with chat.openai.com so uses DOM as the description states. Only recently are API methods slowly being added (like chatgpt.getChatData()) which rely on the access token provided to client when logged in to chat.openai.com. So a method to send prompts hasn't been implemented yet, but it is in the works (if you goto https://github.com/kudoai/duckduckgpt , it will work like that getShowReply(prompt), except you would type chatgpt.send(prompt))

**Comment by thegodone** [comment]
can we extend current solution to be compatible with chatGPT via azure connection : https://learn.microsoft.com/en-us/azure/ai-services/openai/reference ?
